packages/smooth_app/lib/pages/scan/search_panel.dart

```diff
@@ -129,6 +136,29 @@ class SearchPanelState extends State<SearchPanel> {
     });
   }
 
+  void _handleFocusChange() {
+    if (_searchFieldFocusNode.hasFocus) {
+      _controller.open();
+    } else {
+      _controller.close();
+    }
+  }
+
+  void _handleTextChange() {
+    if (_searchFieldController.text.isEmpty && !_isEmpty) {
+      _controller.close();
+    }
```

Comment:
Does this cause the keyboard to stay open

Reply:
No, this also closes the keyboard because of the `unfocus()` call in `_handlePanelSlide()`.